src/nyaml/nyaml2nxdl.py

```diff
@@ -77,29 +56,73 @@
 CATEGORY = ""  # Definition would be either 'base' or 'application'
 
 
-def check_for_dom_comment_in_yaml():
-    """Check the yaml file has dom comment or dom comment needed to be hard coded."""
+def get_nxdl_copyright_license(nxdl_file):
+    """Extract the license part from nxdl file if nxdl file as input."""
+    copyright_lincense = ""
+    comment_start_sym = "^<!--"
+    comment_end_sym = "-->\n+$"
+    is_comment_start = False
+    is_comment_end = False
+
+    if os.path.isfile(nxdl_file):
```

Comment:
Be careful! This function gets the copyright section from a given file. Note that during the conversion
- nxdl->nyaml: we just copy the whole nxdl file to the end of the yaml file
- nyaml->nxdl: we need to generate the copyright from the content of the original nxdl(!) (which is actually appended to the end of the nyaml file).

Reply:
Yeah, this is exactly what is happening. The `nxdl_file` is coming the copied/stored ( in yaml file) nxdl content from `yaml` file. In function `generate_nxdl_or_retrieve_nxdl` `cli.py` that returns the copied/stored `nxdl` part as a output file if there is no changes in the `yaml` part. Otherwise it send the copied/stored `nxdl` content to the `nyaml2nxdl`, that is overwriten later on.
So, the `copyright license` content is coming from `stored` nxdl in yaml file.

You can check this function https://github.com/FAIRmat-NFDI/nyaml/blob/copyright_year/src/nyaml/cli.py#L46.